tests/test_special_ops.py

```diff
@@ -445,6 +445,106 @@ def test_accuracy_unique(shape, dtype, sorted, return_inverse, return_counts):
     gems_assert_equal(res_out, ref_out)
 
 
+UNIQUE_CONSECUTIVE_SHAPE = [
+    # (1024, 65536),
+    # (10000, 65536),
+    (4096, 4096),
+]
+
+
+@pytest.mark.unique_consecutive
+@pytest.mark.parametrize("shape", SPECIAL_SHAPES + UNIQUE_CONSECUTIVE_SHAPE)
+@pytest.mark.parametrize("dtype", INT_DTYPES + FLOAT_DTYPES)
+@pytest.mark.parametrize("return_inverse", [True, False])
+@pytest.mark.parametrize("return_counts", [True, False])
+@pytest.mark.parametrize("dim", [None, 0])
+def test_accuracy_unique_consecutive(shape, dtype, return_inverse, return_counts, dim):
+    if dtype in FLOAT_DTYPES:
+        flat_size = torch.tensor(shape).prod().item()
+        indices = torch.arange(flat_size, device="cuda") // 3
+        unique_vals = torch.randn(indices.max() + 1, dtype=dtype, device="cuda")
+        inp_flat = unique_vals[indices]
+        inp = inp_flat.reshape(shape).to(flag_gems.device)
+    else:
+        flat_size = torch.tensor(shape).prod().item()
+        indices = torch.arange(flat_size, device="cuda") // 2
+        unique_vals = torch.randint(
+            -10, 10, (indices.max() + 1,), device="cuda", dtype=dtype
+        )
+        inp_flat = unique_vals[indices]
+        inp = inp_flat.reshape(shape).to(flag_gems.device)
+
+    ref_inp = to_reference(inp, False)
+
+    if return_counts:
+        if return_inverse:
+            with flag_gems.use_gems():
+                res_output, res_inverse_indices, res_counts = torch.unique_consecutive(
+                    inp,
+                    return_inverse=return_inverse,
+                    return_counts=return_counts,
+                    dim=dim,
+                )
+            ref_output, ref_inverse_indices, ref_counts = torch.unique_consecutive(
+                ref_inp,
+                return_inverse=return_inverse,
+                return_counts=return_counts,
+                dim=dim,
+            )
+            # assert res_output.numel() == ref_output.numel()
+            gems_assert_equal(res_inverse_indices, ref_inverse_indices)
+        else:
+            with flag_gems.use_gems():
+                res_output, res_counts = torch.unique_consecutive(
+                    inp,
+                    return_inverse=return_inverse,
+                    return_counts=return_counts,
+                    dim=dim,
+                )
+            ref_output, ref_counts = torch.unique_consecutive(
+                ref_inp,
+                return_inverse=return_inverse,
+                return_counts=return_counts,
+                dim=dim,
+            )
+            # assert res_output.numel() == ref_output.numel()
+        gems_assert_equal(res_counts, ref_counts)
```

Comment:
Better duplicate this assertion after line 494 and line 510.
The current indentation hurts the readability of the code.